### DISCUSSION TITLE
Empty top languages card

### DISCUSSION THREAD

**Discussion by a-menshchikov** [anuraghazra/github-readme-stats#1789] [Q&A] [answered]
I have an empty Top Languages (https://github-readme-stats.vercel.app/api/top-langs/?username=a-menshchikov):
![image](https://user-images.githubusercontent.com/2580489/157692642-2836b255-3670-4e53-b825-f3d781a88342.png)

Could you explain why?

Previously I have thought that it's a temporary bug, but I see it more than 6 months.

**Comment by rickstaa** [comment, marked as answer]
@a-menshchikov Thanks for your issue. I checked your problem and it seems that something goes wrong while trying to fetch language data for your username using the GraphQL API. You can see this in action by using the following query in the [graphql explorer](https://docs.github.com/en/graphql/overview/explorer):

```graphql
query userInfo($login: String!) {
  user(login: $login) {
    # fetch only owner repos & not forks
    repositories(ownerAffiliations: OWNER, isFork: false, first: 100) {
      nodes {
        name
        languages(first: 10, orderBy: {field: SIZE, direction: DESC}) {
          edges {
            size
            node {
              color
              name
            }
          }
        }
      }
    }
  }
}
```

You can then use the following in the query variables field.

```graphql
{"login": "a-menshchikov"}
```

Fetching the user stats for your account doesn't cause any problems:

```graphql
query userInfo($login: String!) {
  user(login: $login) {
    repositories(first: 100, orderBy: {direction: DESC, field: STARGAZERS}) {
      totalCount
      nodes {
        name
        stargazers {
          totalCount
        }
      }
    }
  }
}
```

However, I noticed that you only have forks after looking through [your repositories](https://github.com/a-menshchikov?tab=repositories). We currently do not fetch data from forks only from parent repositories. This behaviour can be seen by changing the `isFork` parameter in the query to `true`, after which the language stats your looking for show up.

**Reply by rickstaa** [reply]
@a-menshchikov Does the answer above solve your problem?

**Reply by a-menshchikov** [reply]
@rickstaa It helped me to understand that there is no problem. =)

**Comment by rickstaa** [comment]
I agree that this behaviour should be documented. A feature request for also counting language information for code that is found in forks can be found in #1437. I will close this issue for now. Feel free to comment if you still need help.

**Comment by rickstaa** [comment]
Related to #1790.
